api/agent/agent.go

```diff
@@ -101,6 +101,9 @@ type Agent interface {
 	// Enqueue is to use the agent's sweet sweet client bindings to remotely
 	// queue async tasks and should be removed from Agent interface ASAP.
 	Enqueue(context.Context, *models.Call) error
+
+	GetAppByID(ctx context.Context, appID string) (*models.App, error)
+	GetApp(ctx context.Context, app *models.App) (*models.App, error)
```

Comment:
What's the purpose of this one?  If it's to get it by name, why not have `GetAppByName`?